setup.py

```diff
@@ -198,7 +198,7 @@ def run(self):
                 'pydocstyle>=5.1.1',
                 'pytest-cov>=2.11.1',
                 'pytest-subtests>=0.4.0',
-                'pytest>=6.2.2',
+                'pytest>=6.2.2, <=7.4.4',
```

Comment:
It's strange to set the upper version.

Reply:
Pytest version 8.2.2 introduced an issue (https://github.com/pytest-dev/pytest/issues/12289) that broke Knack (Issue: https://github.com/microsoft/knack/issues/287). As a result, the pipeline job for Python 3.8 fails due to this issue:
![image](https://github.com/user-attachments/assets/aae0c5ff-b5c6-4a74-b5df-659b7c1998e3)
I have contacted the Knack team to address this issue. Limiting the pytest version to 7.4.4 (align with python3.7) is a temporary workaround to ensure the SB pipeline passes until Knack provides a fix.